app/vmagent/remotewrite/pendingseries_test.go

```diff
@@ -32,17 +33,19 @@ func testPushWriteRequest(t *testing.T, rowsCount, expectedBlockLenProm, expecte
 			pushBlockLen = len(block)
 		}
 		pushWriteRequest(wr, pushBlock, isVMRemoteWrite)
-		if pushBlockLen != expectedBlockLen {
-			t.Fatalf("unexpected block len for rowsCount=%d, isVMRemoteWrite=%v; got %d bytes; expecting %d bytes",
-				rowsCount, isVMRemoteWrite, pushBlockLen, expectedBlockLen)
+
+		if (tolerancePrc == 0 && pushBlockLen != expectedBlockLen) ||
+			math.Abs(float64(pushBlockLen-expectedBlockLen)/float64(expectedBlockLen)*100) > tolerancePrc {
+			t.Fatalf("unexpected block len for rowsCount=%d, isVMRemoteWrite=%v; got %d bytes; expecting %d bytes +- %.0f%%",
+				rowsCount, isVMRemoteWrite, pushBlockLen, expectedBlockLen, tolerancePrc)
```

Comment:
nit: just to simplify things

```suggestion
if pushBlockLen == expectedBlockLen {
	return
}

if tolerancePrc == 0 {
	t.Fatalf("unexpected block len for rowsCount=%d, isVMRemoteWrite=%v; got %d bytes",
	rowsCount, isVMRemoteWrite, pushBlockLen, expectedBlockLen)
}

if math.Abs(float64(pushBlockLen-expectedBlockLen)/float64(expectedBlockLen)*100) > tolerancePrc {
	t.Fatalf("unexpected block len for rowsCount=%d, isVMRemoteWrite=%v; got %d bytes; expecting %d bytes +- %.0f%%",
	rowsCount, isVMRemoteWrite, pushBlockLen, expectedBlockLen, tolerancePrc)
}
```

Reply:
> Would be nice to explain in more detail why we're having this change in the commit message.
> Although, I don't know is 15% of tolerance is something acceptable.

To tell the truth, i am not at all sure about the need to check the size of the compressed message in tests, just because the compression algorithm is constantly change and theoretically may depend on the environment in which it is launched.
Now the value of is selected empirically, such a solution was proposed yesterday by @valyala

Reply:
Ok, thanks! Let's wait for a review from @valyala or @f41gh7, then.

Reply:
Any way it'll be flaky test and checking pushBlockLen doesn't make much sense, as Amper mentioned compression rate may be different.

 Let's merge it as is.